demo.py

```diff
@@ -1,252 +1,56 @@
-                'font_color_tuple': ('Gray', '1000'),
-                'style': 'Button',
-                },
-            'FlatIconButton': {
-                'color_tuple': ('LightGreen', '500'),
-                'font_color_tuple': ('Gray', '1000'),
-                'style': 'Button',
-                'icon_color_tuple': ('Gray', '1000')
-                },
-            'FlatToggleButton': {
-                'color_tuple': ('LightGreen', '500'),
-                'font_color_tuple': ('Gray', '1000'),
-                'style': 'Button',
-                },
-            'RaisedFlatToggleButton': {
-                'color_tuple': ('LightGreen', '500'),
-                'font_color_tuple': ('Gray', '1000'),
-                'style': 'Button',
-                },
-            'FlatCheckBox': {
-                'color_tuple': ('Gray', '0000'),
-                'check_color_tuple': ('LightGreen', '500'),
-                'outline_color_tuple': ('Gray', '1000'),
-                'style': 'Button',
-                'check_scale': .7,
-                'outline_size': '10dp',
-                },
-            'FlatCheckBoxListItem': {
-                'font_color_tuple': ('Gray', '1000'),
-                'check_color_tuple': ('LightGreen', '500'),
-                'outline_color_tuple': ('Gray', '800'),
-                'style': 'Button',
-                'check_scale': .7,
-                'outline_size': '10dp',
-                },
-            }
-
-        self.theme_manager.add_theme('green', 'main', main)
-        self.theme_manager.add_theme('green', 'accent', accent)
 
+import sys
+sys.path.append('flat_kivy_extensions/submodules')
+
+from flat_kivy_extensions.flatappextension import ExtendedFlatApp
+
+# 1) Developer specifies application title and a string to describe
+#    the application
+
+title = 'My App Title'
+about = 'Something to describe the purpose of the application.'
+
+# 2) Developer specifies what screens should be included in
+#    the screenmanger and how they should be referenced
+#    in the navigation panel for switching
+
+from screens import FlatKivyDemoScreen, KivyScreen1, KivyScreen2
+
+#    An application configuration list is used to specify what screens
+#    should be included and how to navigate to them.
+#
+#    This configuration list defines the widgets that are included in
+#    the navigation side-panel and each item in the list is used to configure
+#    either a:
+#
+#     - Label used for navigation panel separator headers
+#
+#     - Button for each Screen to include.
+#         Each button instance holds a reference to the screen
+#         and triggers navigation to that screen via on_release
+
+# Each item in the app_config_entries list must be one of the following types:
+#
+#  str():  Text used in a Label that is added to th navigation slide-out panel
+#
+#  dict(): Dict of configuration properties that is used to construct
+#          a Label that is added to the navigation slide out-panel
+#
```

Comment:
I think we should consider some type of less tupley setup for the below code. In my experience tuples come back to bite you after enough time. They're hard to read and require humans to explicitly unpack what everything means. It might be a better job to use a type of object here. something like

`
class AppConfig(object):
    def __init__(self, button_title, screen_class, screen_args=None, screen_kwargs=None):
        self.button_title = button_tile
        self.screen_class = screen_class
        if screen_args is None:
             self.screen_args = []
        else:
             self.screen_args = screen_args
        if screen_kwargs is None:
             self.screen_kwargs = {}
        else:
             self.screen_kwargs = screen_kwargs
`

Reply:
Yes - in a very old incarnation I was doing this but some of our developers found it a bit clunky to construct the AppConfig object, even though it makes the framework side of the code cleaner.

This is also useful if we want to add something like specifying icons that can be tied to screen navigation; i've learned the hard way about fixing things like these tuples or assumed list contents, particularly when it comes time to ammend the tuple/list/....

I will adapt to this sooner than later :-)